src/transformers/models/deepseek_v3/modular_deepseek_v3.py

```diff
@@ -153,10 +181,114 @@ def moe(self, hidden_states: torch.Tensor, topk_indices: torch.Tensor, topk_weig
         return final_hidden_states.type(hidden_states.dtype)
 
 
+def _remove_inference_params(module: nn.Module):
+    assert isinstance(module, DeepseekV3Attention)  # Sanity check
+    module.inference_v_proj = None
+    module.inference_q_decode = None
+
+
+def _full_backward_hook(module: nn.Module, grad_input, grad_output):
+    _remove_inference_params(module)
+
+
+def _load_state_dict_post_hook(module: nn.Module, incompatible_keys):
+    _remove_inference_params(module)
```

Comment:
these should / can be ignore with our `_keys_to_ignore_on_load_missing` and `_keys_to_ignore_on_load_unexpected`

Reply:
You sure know this better. I need to detect when weights are changed in any way. I know that backward and load_state_dict will change them, plus there is this fingerprint test.